src/validation.h

```diff
@@ -744,6 +745,62 @@ bool InvalidateBlock(CValidationState& state, const CChainParams& chainparams, C
 /** Remove invalidity status from a block and its descendants. */
 void ResetBlockFailureFlags(CBlockIndex* pindex) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
+/**
+ * Serialized version of a UTXO set from which an assumed-valid CChainState
+ * can be constructed.
+ */
+class SnapshotMetadata
+{
+public:
+    //! The block header of the block that reflects the tip of the chain for the
+    //! UTXO set contained in this snapshot.
+    CDiskBlockIndex m_base_blockheader;
```

Comment:
Why does this need the header (as opposed to just the hash of the header?)

Reply:
During snapshot activation we use `m_base_blockheader.nHeight` to index into the expected assumeutxo hash values, but beyond that I don't think we use other parts of the header. I can split out the height if you like, but I don't see much harm in taking on a few extra bytes for the complete header in case it's somehow advantageous to have in the future.

Reply:
How is `m_base_blockheader.nHeight` different from `LookupBlockIndex(m_base_blockhash)->nHeight`?

Reply:
Ah yeah, that's a good point. I'll revert to using a hash in the snapshot and just do the lookup during activation.

Reply:
thx!